File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.py
import sys

t = int(sys.stdin.readline())

for _ in range (t) : 
    n = int(sys.stdin.readline())
    
    clothes = dict()
    
    for i in range (n) : 
        name, sort = sys.stdin.readline().rstrip().split()
        
        if sort in clothes.keys() :
            clothes[sort].append(name)
        
        else : 
            clothes[sort] = [name]
            
    list_length = [len(compo) for compo in clothes.values()]
   
    counting = 1
    
    for k in list_length : 
        counting *= (k + 1)
        
    counting -= 1
    
    print (counting)

File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int t;
    cin >> t;
    
    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
        
        unordered_map<string, vector<string>> clothes;
        
        for (int j = 0; j < n; j++) {
            string name, sort;
            cin >> name >> sort;
            
            clothes[sort].push_back(name);
        }
        
        vector<int> list_length;
        for (auto& kv : clothes) {
            list_length.push_back(kv.second.size());
        }
        
        int counting = 1;
        for (int len : list_length) {
            counting *= (len + 1);
        }
        
        counting -= 1;
        
        cout << counting << endl;
    }
    
    return 0;
}
